### PR TITLE
feat(cbr/vault): add a new vaults data source support for cbr service

### LINKED ISSUE #1677: [CBR] Want a new data source to obtain the CBR vault list
Hi team,

I want a new data source to obtain the CBR vault list.
1. obtain all vault instances of the special region.
2. obtain all resources of the special vault.

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
Currently, the terraform don't have the data source to query the list of VBR vaults.

**Which issue this PR fixes**:
fixes #1677

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add a new vault data source support.
2. add related test and document.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/cbr' TESTARGS='-run=TestAccCbrVaultsV3_BasicServer'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cbr -v -run=TestAccCbrVaultsV3_BasicServer -timeout 360m -parallel 4
=== RUN   TestAccCbrVaultsV3_BasicServer
=== PAUSE TestAccCbrVaultsV3_BasicServer
=== CONT  TestAccCbrVaultsV3_BasicServer
--- PASS: TestAccCbrVaultsV3_BasicServer (345.71s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cbr       346.933s
```

```
make testacc TEST='./huaweicloud/services/acceptance/cbr' TESTARGS='-run=TestAccCbrVaultsV3_ReplicaServer'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cbr -v -run=TestAccCbrVaultsV3_ReplicaServer -timeout 360m -parallel 4
=== RUN   TestAccCbrVaultsV3_ReplicaServer
=== PAUSE TestAccCbrVaultsV3_ReplicaServer
=== CONT  TestAccCbrVaultsV3_ReplicaServer
--- PASS: TestAccCbrVaultsV3_ReplicaServer (37.60s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cbr       37.716s
```

```
make testacc TEST='./huaweicloud/services/acceptance/cbr' TESTARGS='-run=TestAccCbrVaultsV3_BasicVolume'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cbr -v -run=TestAccCbrVaultsV3_BasicVolume -timeout 360m -parallel 4
=== RUN   TestAccCbrVaultsV3_BasicVolume
=== PAUSE TestAccCbrVaultsV3_BasicVolume
=== CONT  TestAccCbrVaultsV3_BasicVolume
--- PASS: TestAccCbrVaultsV3_BasicVolume (72.77s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cbr       72.896s
```

```
make testacc TEST='./huaweicloud/services/acceptance/cbr' TESTARGS='-run=TestAccCbrVaultsV3_BasicTurbo'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cbr -v -run=TestAccCbrVaultsV3_BasicTurbo -timeout 360m -parallel 4
=== RUN   TestAccCbrVaultsV3_BasicTurbo
=== PAUSE TestAccCbrVaultsV3_BasicTurbo
=== CONT  TestAccCbrVaultsV3_BasicTurbo
--- PASS: TestAccCbrVaultsV3_BasicTurbo (428.67s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cbr       429.092s
```

```
make testacc TEST='./huaweicloud/services/acceptance/cbr' TESTARGS='-run=TestAccCbrVaultsV3_ReplicaTurbo'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cbr -v -run=TestAccCbrVaultsV3_ReplicaTurbo -timeout 360m -parallel 4
=== RUN   TestAccCbrVaultsV3_ReplicaTurbo
=== PAUSE TestAccCbrVaultsV3_ReplicaTurbo
=== CONT  TestAccCbrVaultsV3_ReplicaTurbo
--- PASS: TestAccCbrVaultsV3_ReplicaTurbo (39.93s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cbr       40.034s
```

### CODE REVIEW

File: docs/data-sources/cbr_vaults.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+## Example Usage
+
+### Obtain the vault list by vault name
+
+```hcl
+variable "vault_name" {}
+
+data "huaweicloud_cbr_vaults" "test" {
+  name = var.vault_name
```
Comment (chengxiangdong) [inline docs/data-sources/cbr_vaults.md]: how about this:
`
data "huaweicloud_cbr_vaults" "test" {
  name = "your_cbr_vault_name"
}
`

File: huaweicloud/services/cbr/data_source_huaweicloud_cbr_vaults.go [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+			},
+			"enterprise_project_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"policy_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"status": {
```
Comment (chengxiangdong) [inline huaweicloud/services/cbr/data_source_huaweicloud_cbr_vaults.go:73]: `status` is not described in the doc

File: huaweicloud/services/cbr/data_source_huaweicloud_cbr_vaults.go [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+	pages, err := vaults.List(client, opt).AllPages()
+	if err != nil {
+		return fmtp.DiagErrorf("Error getting vault details: %s", err)
+	}
+	resp, err := vaults.ExtractVaults(pages)
+
+	// Use the following parameters to filter the result of the List method return: consistent_level, size, auto_expand_enabled.
+	vaultList, err := filterCbrVaults(d, *resp)
+	if err != nil {
+		return fmtp.DiagErrorf("")
```
Comment (chengxiangdong) [inline huaweicloud/services/cbr/data_source_huaweicloud_cbr_vaults.go]: is it not finished yet?

File: huaweicloud/services/cbr/data_source_huaweicloud_cbr_vaults.go [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+	}
+
+	// Set the ID and other parameters.
+	err = setCbrAllVaultParameters(client, d, vaultList)
+	if err != nil {
+		return fmtp.DiagErrorf("")
```
Comment (chengxiangdong) [inline huaweicloud/services/cbr/data_source_huaweicloud_cbr_vaults.go]: ditto

File: huaweicloud/services/cbr/data_source_huaweicloud_cbr_vaults.go (hunk 2 above)
Comment (Lance52259) [reply huaweicloud/services/cbr/data_source_huaweicloud_cbr_vaults.go:73]: yeah, It has been added.

File: huaweicloud/services/cbr/data_source_huaweicloud_cbr_vaults.go (hunk 3 above)
Comment (Lance52259) [reply huaweicloud/services/cbr/data_source_huaweicloud_cbr_vaults.go]: That seems to be the case. =-=
The return messsage has been updated.

File: huaweicloud/services/cbr/data_source_huaweicloud_cbr_vaults.go (hunk 4 above)
Comment (Lance52259) [reply huaweicloud/services/cbr/data_source_huaweicloud_cbr_vaults.go]: done.

File: docs/data-sources/cbr_vaults.md (hunk 1 above)
Comment (Lance52259) [reply docs/data-sources/cbr_vaults.md]: To be debated.

File: docs/data-sources/cbr_vaults.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+* `policy_id` - (Optional, String) Specifies a policy to associate with the CBR vault.
+  The `policy_id` cannot be used with the vault of replicate protection type.
+
+* `status` - (Optional, String) Specifies the CBR vault status, including **available**, **lock**, **frozen** and **error**.
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The data source ID in UUID format.
```
Comment (hmbig2) [inline docs/data-sources/cbr_vaults.md]: hashcode is not UUID

Comment (Lance52259) [reply docs/data-sources/cbr_vaults.md]: OFC, edited

File: docs/data-sources/cbr_vaults.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+  + **turbo** (SFS Turbo file systems)
+
+* `consistent_level` - (Optional, String) Specifies the backup specifications.
+  The valid values are as follows:
+  + **[crash_consistent](https://support.huaweicloud.com/intl/en-us/usermanual-cbr/cbr_03_0109.html)**
+  + **[app_consistent](https://support.huaweicloud.com/intl/en-us/usermanual-cbr/cbr_03_0109.html)**
+
+  Only server type vaults support application consistent.
+
+* `protection_type` - (Optional, String, ForceNew) Specifies the protection type of the CBR vault.
```
Comment (niuzhenguo) [inline docs/data-sources/cbr_vaults.md]: `ForceNew` is unneeded here.

Comment (Lance52259) [reply docs/data-sources/cbr_vaults.md]: The `ForceNew` has been removed.

**Pull request merged**